tools/src/test/scala/scala/scalanative/NIRCompiler.scala

```diff
@@ -148,3 +112,58 @@ object NIRCompiler {
   }
 
 }
+
+class NIRCompilerImpl(outDir: Path) extends api.NIRCompiler {
+
+  def this() = this(Files.createTempDirectory("scala-native-target"))
+
+  override def compile(base: Path): Array[Path] = {
+    val files = getFiles(base.toFile(), _.getName endsWith ".scala")
+    println(files)
+    compile(files)
+  }
+
+  override def compile(source: String): Array[Path] = {
+    val tempFile = File.createTempFile("scala-native-input", ".scala").toPath
+    Files.write(tempFile, source.getBytes(StandardCharsets.UTF_8))
+    compile(Seq(tempFile.toFile()))
+  }
+
+  private def compile(files: Seq[File]): Array[Path] = {
+    val mainClass = if (ScalaNativeBuildInfo.scalaVersion.startsWith("3")) "dotty.tools.dotc.Main" else "scala.tools.nsc.Main"
+    val outPath = outDir.toAbsolutePath()
+    val args = files.map(_.getAbsolutePath()).mkString(",")
+    val procBuilder = new ProcessBuilder(
+      "java",
+      "-cp", s"\"${ScalaNativeBuildInfo.scalacJars}\"",
+      mainClass,
+      "-d", s"$outPath",
+      s"-Xplugin:${ScalaNativeBuildInfo.pluginJar}",
+      args
+    )
+    val cmd = List(
+      "java",
+      "-cp", s"\"${ScalaNativeBuildInfo.scalacJars}\"",
+      mainClass,
+      "-d", s"$outPath",
+      s"-Xplugin:${ScalaNativeBuildInfo.pluginJar}",
+      args
+    ).mkString(" ")
+    println(cmd)
```

Comment:
What is missing here is the `-cp` for the Scala compiler. Even though it might couner intutive, currently we only have passed classpath to the java runtime. Instead, we should have first -cp with full classpath required to run scala-compiler, and second `-cp` in the arguments list describing the classpath of the compilation unit for the Scala comoiler. As a reference we can take a look at outputs of `scala-cli compile --server=false main.scala`: 

```bash 
/Users/wmazur/.sdkman/candidates/java/current/bin/java
-cp # classpath for java runtime
/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-compiler_3/3.3.0/scala3-compiler_3-3.3.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-interfaces/3.3.0/scala3-interfaces-3.3.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.3.0/scala3-library_3-3.3.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/tasty-core_3/3.3.0/tasty-core_3-3.3.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-asm/9.4.0-scala-1/scala-asm-9.4.0-scala-1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/compiler-interface/1.3.5/compiler-interface-1.3.5.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-reader/3.19.0/jline-reader-3.19.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal/3.19.0/jline-terminal-3.19.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal-jna/3.19.0/jline-terminal-jna-3.19.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.10/scala-library-2.13.10.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.7.0/protobuf-java-3.7.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/util-interface/1.3.0/util-interface-1.3.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/net/java/dev/jna/jna/5.3.1/jna-5.3.1.jar 
dotty.tools.dotc.Main # runtime arguments starting with main class 
-sourceroot
/Users/wmazur/projects/scala-native/sandbox/src/main/scala
-release
17
-d
/Users/wmazur/projects/scala-native/sandbox/src/main/scala/.scala-build/project_103be31561/classes/main
-cp # classpath for Scala compiler
/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.3.0/scala3-library_3-3.3.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.10/scala-library-2.13.10.jar
/Users/wmazur/projects/scala-native/sandbox/src/main/scala/Test.scala

```

Reply:
To sum things up we should have these things in the Scala compiler -cp: 
* Scala 2 stdlib jar
* Scala 3 stdlib jar (optionally) 
* Scala Native nativelib jar

Potentially we could get all of these using `nativelib/fullClasspathAsJars` task

Reply:
Thanks @WojciechMazur !
I could make it work in my local env 👍